### ISSUE TITLE
failed to access when deploy waypoint with ns granularity

### ISSUE THREAD

**Issue by YaoZengzeng** [kmesh-net/kmesh#628]
**What happened**:

I deploy bookinfo, sleep pod as client and waypoint with ns granularity, ref https://kmesh.net/en/docs/userguide/try_waypoint/ to reproduce.

Then I failed to access bookinfo:

```bash
root@kurator-linux-0002:~/kmesh# kubectl exec deploy/sleep -- curl -s http://productpage:9080/ | grep -o "<title>.*</title>"
command terminated with exit code 1
```

The log of Kmesh daemon is:

```bash
time="2024-07-23T06:30:36Z" level=info msg="[BACKEND] ERR: record metadata origin address and port failed, ret is 0\n" subsys=ebpf
time="2024-07-23T06:30:36Z" level=info msg="[BACKEND] ERR: waypoint_manager failed, ret:0\n" subsys=ebpf
time="2024-07-23T06:30:36Z" level=info msg="[FRONTEND] ERR: service_manager failed, ret:0\n" subsys=ebpf
time="2024-07-23T06:30:36Z" level=info msg="[KMESH] ERR: frontend_manager failed, ret:0\n" subsys=ebpf
time="2024-07-23T06:30:36Z" level=info msg="[KMESH] ERR: sock_traffic_control failed: 0\n" subsys=ebpf
time="2024-07-23T06:30:36Z" level=info msg="[PROBE] ERR: connect bpf_sk_storage_get failed\n" subsys=ebpf
time="2024-07-23T06:30:36Z" level=info msg="[PROBE] ERR: close bpf_sk_storage_get failed\n" subsys=ebpf
```

But it works fine when the waypoint is service or pod granularity.

**What you expected to happen**:

It should work fine and it should have worked before. After this bug is fixed, the corresponding E2E test cases (https://github.com/kmesh-net/kmesh/pull/625) can be merged.

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kmesh version:
- Others:

**Comment by hzxuzhonghu** [comment]
weired why ret = 0 is treated as error

**Comment by hzxuzhonghu** [comment]
should be fixed by #659
